test-data/unit/check-overloading.test

```diff
@@ -4125,6 +4125,59 @@ main:11: error: Argument 7 to "f" has incompatible type "Union[int, str]"; expec
 main:11: error: Argument 8 to "f" has incompatible type "Union[int, str]"; expected "int"
 main:11: note: Not all union combinations were tried because there are too many unions
 
+[case testSafeDunderOverlapInSubclass]
+from typing import overload
+
+class A:
+    def f(self, x : 'A') -> 'A': ...
```

Comment:
Maybe we should also add a unit test to see what happens if the parent class is also defining an overloaded operator? Up to you.

Reply:
Oh wait, it looks like I used `f` instead of `__add__`, I will switch to `__add__`. Btw the test case you mention already exists in other file, so I will just keep this one.

Reply:
Actually this test gets updated in this PR, it was passing but now fails, see `testOverloadedOperatorMethodOverrideWithNewItem`.